cmd-car-split.go

```diff
@@ -0,0 +1,291 @@
+					}
+				}
+			}
+
+			epoch := c.Int("epoch")
+			maxFileSize := c.Int64("size")
+
+			var (
+				currentFileSize   int64
+				currentFileNum    int
+				currentFile       *os.File
+				fileMutex         sync.Mutex
+				currentSubsetInfo subsetInfo
+				subsetLinks       []datamodel.Link
+			)
+
+			createNewFile := func() error {
+				fileMutex.Lock()
+				defer fileMutex.Unlock()
+
+				if currentFile != nil {
+					subsetNode, err := qp.BuildMap(ipldbindcode.Prototypes.Subset, -1, func(ma datamodel.MapAssembler) {
+						qp.MapEntry(ma, "kind", qp.Int(int64(iplddecoders.KindSubset)))
+						qp.MapEntry(ma, "first", qp.Int(int64(currentSubsetInfo.firstSlot)))
+						qp.MapEntry(ma, "last", qp.Int(int64(currentSubsetInfo.lastSlot)))
+						qp.MapEntry(ma, "blocks",
+							qp.List(-1, func(la datamodel.ListAssembler) {
+								for _, bl := range currentSubsetInfo.blockLinks {
+									qp.ListEntry(la, qp.Link(bl))
+								}
+							}))
+					})
+					if err != nil {
+						return err
+					}
+
+					cid, err := writeNode(subsetNode, currentFile)
+					if err != nil {
+						return err
+					}
+
+					subsetLinks = append(subsetLinks, cidlink.Link{Cid: cid})
+
+					currentFile.Close()
+				}
+				currentFileNum++
+				filename := fmt.Sprintf("epoch-%d-%d.car", epoch, currentFileNum)
+				currentFile, err = os.Create(filename)
+				if err != nil {
+					return err
+				}
+				// Write the header
+				_, err = io.WriteString(currentFile, nulRootCarHeader)
+				if err != nil {
+					return err
+				}
+
+				// Set the currentFileSize to the size of the header
+				currentFileSize = int64(len(nulRootCarHeader))
+				currentSubsetInfo = subsetInfo{fileName: filename, firstSlot: -1, lastSlot: -1}
+				return nil
+			}
+
+			writeObject := func(data []byte) error {
+				fileMutex.Lock()
+				defer fileMutex.Unlock()
+
+				if currentFile == nil || currentFileSize+int64(len(data)) > maxFileSize {
+
+					if err := createNewFile(); err != nil {
+						return err
+					}
+				}
+
+				_, err := currentFile.Write(data)
+				if err != nil {
+					return fmt.Errorf("failed to write object to car file: %s, error: %w", currentFile.Name(), err)
+				}
+				currentFileSize += int64(len(data))
+				return nil
+			}
+
+			processObject := func(owm *accum.ObjectWithMetadata) error {
+				data := owm.ObjectData
+				kind, err := iplddecoders.GetKind(data)
+				if err != nil {
+					return err
+				}
+
+				if kind == iplddecoders.KindBlock {
+					block, err := iplddecoders.DecodeBlock(data)
+					if err != nil {
+						return err
+					}
+
+					if currentSubsetInfo.firstSlot == -1 || block.Slot < currentSubsetInfo.firstSlot {
+						currentSubsetInfo.firstSlot = block.Slot
+					}
+					if block.Slot > currentSubsetInfo.lastSlot {
+						currentSubsetInfo.lastSlot = block.Slot
+					}
+
+					currentSubsetInfo.blockLinks = append(currentSubsetInfo.blockLinks, cidlink.Link{Cid: owm.Cid})
+				}
+
+				return writeObject(data)
```

Comment:
this will write only the value; a valid CAR object entry (aka "section") is made of `len + key + value`

There's a method to get that, called `RawSection`.

Reply:
Fixed.